persistent/Database/Persist/Sql/Orphan/PersistUnique.hs

```diff
@@ -23,26 +23,15 @@ import Control.Monad.Trans.Reader (ask, withReaderT, ReaderT)
 import Data.List (nubBy)
 import Data.Function (on)
 
-defaultUpsert
-    :: (MonadIO m
-       ,PersistEntity record
-       ,PersistUniqueWrite backend
-       ,PersistEntityBackend record ~ BaseBackend backend)
-    => record -> [Update record] -> ReaderT backend m (Entity record)
-defaultUpsert record updates = do
-    uniqueKey <- onlyUnique record
-    upsertBy uniqueKey record updates
-
 instance PersistUniqueWrite SqlBackend where
-    upsert record updates = do
+    upsertBy uniqueKey record updates = do
```

Comment:
`upsert` calls `upsertBy` under the hood (and may be deprecated in the future), so providing `upsertBy` here instead of `upsert` should not break anything.